### ISSUE TITLE
Add support for moving foreign key referenced columns between tables

### ISSUE THREAD

**Issue by silentninja** [centerofci/mathesar#1433]
## Problem
Extracting columns into a new table or moving columns to an existing table does not work with columns that are referenced by other columns.
## Proposed solution
Columns moved to a different table should break the foreign key constraint and establish a new constraint that points to the new column.

**Comment by ritikBhandari** [comment]
Are we trying to move column from one table to another?

**Comment by kgodey** [comment]
> Are we trying to move column from one table to another?

@ritikBhandari Yes. Please see https://github.com/centerofci/mathesar/pull/1430 for more detail, this was a todo that came out of that pull request. https://github.com/centerofci/mathesar/pull/1432 is also related.

**Comment by ritikBhandari** [comment]
Sure.

**Comment by kgodey** [comment]
@ritikBhandari Are you planning to submit a pull request that fixes this issue?

**Comment by ritikBhandari** [comment]
Yes I do plan to submit a PR.

**Comment by mathemancer** [comment]
@ritikBhandari Please ensure your work is well-aligned with #1432 . Thank you!

**Comment by ritikBhandari** [comment]
Sure @mathemancer . Where can I communicate for any query regarding the issue? Here only?

**Comment by mathemancer** [comment]
You can either discuss here (preferred) or use https://matrix.to/#/#backend:matrix.mathesar.org for quick questions. That's a link to our backend channel on our matrix server.

**Comment by ritikBhandari** [comment]
Sure @mathemancer . Where can I communicate for any query regarding the issue? Here only?

**Comment by ritikBhandari** [comment]
Great! Thank you.

**Comment by ritikBhandari** [comment]
>Columns moved to a different table should break the foreign key constraint and establish a new constraint that points to the new column.

After we have moved the desired column to a new/existing table, the other column which is referencing to this foreign key column in the old table should reference the same column in the new table right?

**Comment by ritikBhandari** [comment]
> > Columns moved to a different table should break the foreign key constraint and establish a new constraint that points to the new column.
> 
> After we have moved the desired column to a new/existing table, the other column which is referencing to this foreign key column in the old table should reference the same column in the new table right?

Like if the other column was referencing to this Foreign key column in table X1 -> FX1.
Now it should refer to FX2, where X2 is the table where column is moved. Right?

**Comment by kgodey** [comment]
@ritikBhandari Yes, that is correct. Tagging @silentninja in case he has more detail.

**Comment by ritikBhandari** [comment]
is there anything @silentninja has to add?

**Comment by ritikBhandari** [comment]
So the other column should just refer to FX2 right?

**Comment by silentninja** [comment]
> So the other column should just refer to FX2 and there would be no FX1 right?

Exactly!

**Comment by kgodey** [comment]
@ritikBhandari since we haven't heard from you for a while, I'm unassigning you from this issue.

**Comment by kgodey** [comment]
Closing this, we will track column moving improvements and fixes in our roadmap (to be published soon).
